bmc-log-collector/log-collector.go

```
@@ -0,0 +1,134 @@
+	intervalTime    time.Duration // interval (sec) time of scraping
+}
+
+func (c *selCollector) collectSystemEventLog(ctx context.Context, m Machine, logWriter bmcLogWriter) {
+	layout := "2006-01-02T15:04:05Z07:00"
+	var createUnixtime int64
+	var lastId int
+
+	lastPtr, err := readLastPointer(m.Serial, m.NodeIP, c.ptrDir)
+	if err != nil {
+		slog.Error("can't read a pointer file.", "err", err, "serial", m.Serial, "ptrDir", c.ptrDir)
+		return
+	}
+
+	bmcUrl := "https://" + m.BmcIP + c.rfSelPath
+	byteJSON, statusCode, err := requestToBmc(ctx, c.username, c.password, c.httpClient, bmcUrl)
+	if statusCode != 200 || err != nil {
+		if statusCode != 200 {
+			slog.Error("failed access to iDRAC.", "url", c.rfSelPath, "httpStatusCode", statusCode)
+		}
+		// increment the metrics counter
+		counterRequestFailed.WithLabelValues(m.Serial, m.NodeIP).Inc()
+		if lastPtr.LastError != err {
+			slog.Error("failed access to iDRAC.", "err", err, "url", c.rfSelPath, "httpStatusCode", statusCode)
+		}
+		lastPtr.LastError = err
+		err = updateLastPointer(lastPtr, c.ptrDir)
+		if err != nil {
+			slog.Error("failed to write a pointer file.", "err", err, "serial", m.Serial, "createUnixtime", createUnixtime, "LastReadId", lastId, "ptrDir", c.ptrDir)
+		}
+		return
+	}
+
+	// increment the metrics counter
+	counterRequestSuccess.WithLabelValues(m.Serial, m.NodeIP).Inc()
+
+	var members RedfishJsonSchema
+	if err := json.Unmarshal(byteJSON, &members); err != nil {
+		slog.Error("failed to translate JSON to go struct.", "err", err, "serial", m.Serial, "ptrDir", c.ptrDir)
+		return
+	}
+
+	for i := len(members.Sel) - 1; i >= 0; i-- {
+		t, _ := time.Parse(layout, members.Sel[i].Create)
+		createUnixtime = t.Unix()
+		lastId, _ = strconv.Atoi(members.Sel[i].Id)
+		members.Sel[i].Serial = m.Serial
+		members.Sel[i].BmcIP = m.BmcIP
+		members.Sel[i].NodeIP = m.NodeIP
+
+		if lastPtr.LastReadId < lastId {
+			// normal case
+			bmcByteJsonLog, _ := json.Marshal(members.Sel[i])
```

You should not ignore error.

---

Added the error handling code.